meilisearch-auth/src/store.rs

```diff
@@ -213,11 +206,24 @@ impl HeedAuthStore {
         &self,
         uid: Uuid,
         action: Action,
-        index: Option<&[u8]>,
+        index: Option<&str>,
     ) -> Result<Option<Option<OffsetDateTime>>> {
         let rtxn = self.env.read_txn()?;
-        let tuple = (&uid, &action, index);
-        Ok(self.action_keyid_index_expiration.get(&rtxn, &tuple)?)
+        let tuple = (&uid, &action);
+        match self.action_keyid_index_expiration.get(&rtxn, &tuple)? {
+            None => Ok(None),
+            Some(x) => {
+                if let Some(y) = index {
+                    if x.0.into_iter().any(|x| x == *y) {
```

Comment:
I am not sure but using equal here is probably not idiomatic I would prefer that we introduce a `match` method or something, don't you think?

Reply:
x is a struct with StarOr<IndexType> type which is a representation of index name / pattern. It would be appropriate in comparing the x with string.